src/core/geocms/geonode/qgsgeonoderequest.cpp

```diff
@@ -54,18 +54,22 @@ void QgsGeoNodeRequest::abort()
 void QgsGeoNodeRequest::fetchLayers()
 {
   request( QStringLiteral( "/api/layers/" ) );
-  QObject *obj = new QObject( this );
 
+  QObject *obj = new QObject( this );
   connect( this, &QgsGeoNodeRequest::requestFinished, obj, [obj, this ]
   {
-    QList<QgsGeoNodeRequest::ServiceLayerDetail> layers;
-    if ( mError.isEmpty() )
+    if ( !mParsingLayers )
     {
-      layers = parseLayers( this->lastResponse() );
+      mParsingLayers = true;
+      QList<QgsGeoNodeRequest::ServiceLayerDetail> layers;
+      if ( mError.isEmpty() )
+      {
+        layers = parseLayers( this->lastResponse() );
```

Comment:
is `this->` necessary?

Reply:
@elpaso , most likely not, well spotted cleanup task :) I'll modify.